src/nvim/os/job.c

```diff
@@ -92,27 +115,15 @@ void job_teardown(void)
 /// @return The job pointer if the job started successfully, NULL otherwise
 Job *job_start(JobOptions opts, int *status)
 {
-  int i;
-  Job *job;
-
-  // Search for a free slot in the table
-  for (i = 0; i < MAX_RUNNING_JOBS; i++) {
-    if (table[i] == NULL) {
-      break;
-    }
-  }
+  Job *job = xmalloc(sizeof(Job));
 
-  if (i == MAX_RUNNING_JOBS) {
-    // No free slots
-    shell_free_argv(opts.argv);
-    *status = 0;
+  // Initialize
+  job->id = new_job_id();
+  *status = job->id;
+  if (*status == -1) {
```

Comment:
:+1: -- actually, if we can't allocate a job ID, might as well not `xmalloc()` to begin with.